### PR TITLE
refactor(accordion): accordion and collasiple ts to sfc

### PR DESCRIPTION
**Pull request by teleskop150750**
### Linked Issues

cc: https://github.com/oku-ui/primitives/issues/492

### CODE REVIEW

**Comment by productdevbook** [comment]
It's great to see a pr like this, can you get a pr draft if it's not finished? Thank you very, very much. ❤️

File: packages/vue/src/presence/Presence.vue
Comment (productdevbook) [inline packages/vue/src/presence/Presence.vue:1]: './usePresence.js' -> ''./usePresence' no need to type js. The compiler will take care of them.

File: packages/vue/src/provide/createProvide.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ function createScope<T extends string>(scopeName: T, createScopeDeps: CreateScop
    * --------------------------------------------------------------------------------------------- */
 
   function createProvide<ProvideValueType extends object | null>(
-    rootComponentName: T,
+    rootComponentName: string,
```
Comment (productdevbook) [inline packages/vue/src/provide/createProvide.ts:41]: why change string `T` -> `string` ?

Comment (teleskop150750) [reply packages/vue/src/provide/createProvide.ts:41]: in the original it has a ```string``` type

[link](https://github.com/oku-ui/primitives/pull/531#discussion_r1510651446)

AND

fix type error

[link](https://github.com/oku-ui/primitives/blob/b860fd4667d7987d9982562ad00dd09ebda707d8/packages/vue/src/accordion/props.ts#L94)

File: packages/vue/src/presence/Presence.vue
Comment (teleskop150750) [reply packages/vue/src/presence/Presence.vue:1]: OK

I just followed this advice

[link](https://vitejs.dev/guide/performance.html#reduce-resolve-operations)

**Comment by teleskop150750** [comment]
I absolutely cannot understand how it works and why it might be useful.

[link](https://github.com/oku-ui/primitives/blob/b860fd4667d7987d9982562ad00dd09ebda707d8/packages/vue/src/provide/createProvide.ts#L84)

**Comment by teleskop150750** [comment]
> It's great to see a pr like this, can you get a pr draft if it's not finished? Thank you very, very much. ❤️

Плохо понимаю (

**Comment by productdevbook** [comment]
https://github.com/oku-ui/primitives/pull/531 most of the components are broken. They work in storybook. Can you check them? Somewhere there have been fractures.

**Comment by teleskop150750** [comment]
> #492 most of the components are broken. They work in storybook. Can you check them? Somewhere there have been fractures.
> 
> maybe I can send some commits when I have more time.

Я собирался продолжить работу по переносу компонентов на sfc. Большинство компонентов не работают, потому что они импортируют объект props, которого больше нет. Сейчас это interface

**Comment by productdevbook** [comment]
Those with SFC transitions will work.  Try not to break other components when creating PR. Also the storybook does not work in accordion.

**Comment by teleskop150750** [comment]
> Those with SFC transitions will work. Try not to break other components when creating PR. Also the storybook does not work in accordion.

the last commit I was going to make was with the 0.6.1 tag

**Comment by teleskop150750** [comment]
> Those with SFC transitions will work. Try not to break other components when creating PR. Also the storybook does not work in accordion.

the last commit that was built without errors was with the tag 0.6.2

**Comment by productdevbook** [comment]
you made the accordion move sfc in this pr, but the storybook doesn't work at all. You've got destructive changes all over the place. Did you check the storybook accordion while making them?

**Comment by teleskop150750** [comment]
corrected the storybook

**Comment by teleskop150750** [comment]
> you made the accordion move sfc in this pr, but the storybook doesn't work at all. You've got destructive changes all over the place. Did you check the storybook accordion while making them?

corrected the storybook

**Comment by teleskop150750** [comment]
I am currently doing ``tabs``

**Comment by productdevbook** [comment]
LGTM! Thank you for your wonderful work. ❤️

**Review by productdevbook — APPROVED** [review]

**Pull request merged**